src/queue.rs

```diff
@@ -0,0 +1,105 @@
+use std::fmt;
+
+// tmp allow dead_code
+#[allow(dead_code)]
+pub struct Queue<T: Clone> {
+    head: Option<Node<T>>,
+    length: i32,
+}
+
+// tmp allow dead_code
+#[allow(dead_code)]
+struct Node<T: Clone> {
+    data: T,
+    next: Option<Box<Node<T>>>,
+}
+
+// tmp allow dead_code
+#[allow(dead_code)]
+pub fn new<T: Clone>() -> Queue<T> {
+    return Queue {
+        head: None,
+        length: 0,
+    };
+}
+
+pub struct EmptyQueueError;
+impl fmt::Display for EmptyQueueError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Queue is empty")
+    }
+}
+
+// tmp allow dead_code
+#[allow(dead_code)]
+impl<'a, T: Clone> Queue<T> {
```

Comment:
```suggestion
impl<T: Clone> Queue<T> {
```
unused lifetime parameter

Reply:
boo on me